libs/testing/src/jobs.service.ts

```diff
@@ -74,38 +64,29 @@ export class JobsService {
         }
       : {};
 
+    let totalCount = 0;
+
     do {
+      // Wait until Bull queues are empty
       totalCount = (await this.getQueueMetric()).totalCount;
-      runningJobs = await this.jobRepository.count({
-        _organizationId: organizationId,
-        ...typeMatch,
-        ...workflowMatch,
-        status: {
-          $in: [JobStatusEnum.PENDING, JobStatusEnum.QUEUED, JobStatusEnum.RUNNING],
-        },
-      });
-    } while (totalCount > 0 || runningJobs > unfinishedJobs);
+      // Wait until there are no pending, queued or running jobs in Mongo
+      runningJobs = Math.max(
+        await this.jobRepository.count({
+          _organizationId: organizationId,
+          ...typeMatch,
+          ...workflowMatch,
+          status: {
+            $in: [JobStatusEnum.PENDING, JobStatusEnum.QUEUED, JobStatusEnum.RUNNING],
+          },
+        }),
+        0
+      );
+    } while (totalCount > 0 || runningJobs > safeUnfinishedJobs);
+  }
 
-    return {
-      getDelayedTimestamp: async () => {
-        const delayedJobs = await this.standardQueue.getDelayed();
-
-        if (delayedJobs.length === 1) {
-          return delayedJobs[0].delay;
-        } else if (delayedJobs.length > 1) {
-          throw new Error('There are more than one delayed jobs');
-        } else if (delayedJobs.length === 0) {
-          throw new Error('There are no delayed jobs');
-        }
-      },
-      runDelayedImmediately: async () => {
-        const delayedJobs = await this.standardQueue.getDelayed();
-
-        await delayedJobs.forEach(async (job) => {
-          job.changeDelay(1);
-        });
-      },
-    };
+  public async runAllDelayedJobsImmediately() {
+    const delayedJobs = await this.standardQueue.getDelayed();
```

Comment:
i thought we used this more nested in the logic, but now i see that it was used only once in the tests.
i am not sure that i like this name because it is misleading, what it will do is `run the current queued delay job immediately` and not `run all..`
for example, if you will have workflow: in-app-1 -> delay-1 -> in-app-2 -> delay-2 -> in-app-3
and you run 

```
await session.waitForJobCompletion(template?._id);
await session.runAllDelayedJobsImmediately();
```
    
you could still have jobs in-app-2, delay-2, in-app-3 pending in the pipeline.

Reply:
Good call out. I will update the name.